src/components/ButtonV2/index.tsx

```diff
@@ -0,0 +1,238 @@
+      'variant',
+      'size',
+      'prominence',
+      'extend',
+      'as',
+      'target',
+    ].includes(prop),
+})<StyledButtonProps>`
+  display: inline-flex;
+  align-items: center;
+  gap: ${({ theme }) => theme.space[1]};
+
+  font-weight: 500;
+  font-size: 16px;
+  border-radius: 4px;
+
+  ${({ size }) => sizes[size]};
+
+  ${({ theme, disabled, variant, prominence, extend }) => {
+    const backgroundValue: keyof typeof theme.colors.primary = `background${
+      prominence === 'strong' ? 'Strong' : 'Weak'
+    }${disabled ? 'Disabled' : ''}`
+    const colorValue: keyof typeof theme.colors.primary = `text${
+      prominence === 'strong' ? 'Strong' : 'Weak'
+    }${disabled ? 'Disabled' : ''}`
+
+    return `
+      background: ${
+        variant === 'neutral'
+          ? theme.colors.neutral.background
+          : theme.colors[variant][backgroundValue]
+      };
+      color: ${
+        variant === 'neutral'
+          ? theme.colors.neutral.text
+          : theme.colors[variant][colorValue]
+      };
+      border: ${
+        variant === 'neutral' || prominence === 'weak'
+          ? `1px solid ${theme.colors[variant].borderWeak}`
+          : 'none'
+      };
+      
+      ${
+        !disabled
+          ? `
+        cursor: pointer;
+        &:hover,:focus-visible {
+          outline : 0;
+          background: ${
+            variant === 'neutral'
+              ? theme.colors.neutral.backgroundWeakHover
+              : theme.colors[variant][
+                  `${backgroundValue}Hover` as keyof typeof theme.colors.primary
+                ]
+          };
+          color: ${
+            variant === 'neutral'
+              ? theme.colors.neutral.textHover
+              : theme.colors[variant][
+                  `${colorValue}Hover` as keyof typeof theme.colors.primary
+                ]
+          };
+          border: ${
+            variant === 'neutral' || prominence === 'weak'
+              ? `1px solid ${theme.colors[variant].borderWeakHover}`
+              : 'none'
+          };
+        }
+
+        &:active {
+          outline : 3px solid #4F059940;
+        }
+      `
+          : /* Disabled */ `
+         
+          color : ${
+            variant === 'neutral'
+              ? theme.colors.neutral.textDisabled
+              : theme.colors[variant][colorValue]
+          };
+          border: ${
+            variant === 'neutral' || prominence === 'weak'
+              ? `1px solid ${
+                  variant === 'neutral'
+                    ? theme.colors.neutral.borderDisabled
+                    : theme.colors[variant].borderWeakDisabled
+                }`
+              : 'none'
+          };
+          `
+      }
+
+      ${
+        extend
+          ? `
+        transition: max-width 450ms ease;
+        max-width: 48px;
+        [data-text] { display: none; }
+
+        &:hover, &:focus {
+          [data-text] { display: inherit }
+          max-width: 275px;
```

Comment:
What happens with long text in French or Deutsch ?

Reply:
Même souci que Button actuel, mais pas sure qu'il y est une meilleure solution, ou sinon en se servant d'un code JS pour connaitre la taille du contenu et ajuster (si tenté que ce soit jouable). Je vais regarder si je trouve un truc plus flexible mais avoir une animation avec une width auto ca me parrait mort en pur CSS